services/checks/checks_test.go

```diff
@@ -127,6 +127,101 @@ func TestCollectChecks(t *testing.T) {
 	})
 }
 
+func TestDisableChecks(t *testing.T) {
+	err := os.Setenv("PERCONA_TEST_CHECKS_FILE", "../../testdata/checks/checks.yml")
+	require.NoError(t, err)
+	defer os.Unsetenv("PERCONA_TEST_CHECKS_FILE") //nolint:errcheck
+
+	t.Run("normal", func(t *testing.T) {
+		sqlDB := testdb.Open(t, models.SkipFixtures, nil)
+		db := reform.NewDB(sqlDB, postgresql.Dialect, nil)
+		s, err := New(nil, nil, db)
+		require.NoError(t, err)
+		s.collectChecks(context.Background())
+
+		checks := s.GetAllChecks()
+		assert.Len(t, checks, 3)
+
+		disChecks, err := s.GetDisabledChecks()
+		require.NoError(t, err)
+		assert.Empty(t, disChecks)
+
+		err = s.DisableChecks([]string{checks[0].Name})
+		require.NoError(t, err)
+
+		disChecks, err = s.GetDisabledChecks()
+		require.NoError(t, err)
+		assert.Len(t, disChecks, 1)
+	})
+
+	t.Run("disable same check twice", func(t *testing.T) {
+		sqlDB := testdb.Open(t, models.SkipFixtures, nil)
+		db := reform.NewDB(sqlDB, postgresql.Dialect, nil)
+		s, err := New(nil, nil, db)
+		require.NoError(t, err)
+		s.collectChecks(context.Background())
+
+		checks := s.GetAllChecks()
+		assert.Len(t, checks, 3)
+
+		disChecks, err := s.GetDisabledChecks()
+		require.NoError(t, err)
+		assert.Empty(t, disChecks)
+
+		err = s.DisableChecks([]string{checks[0].Name})
+		require.NoError(t, err)
+
+		err = s.DisableChecks([]string{checks[0].Name})
+		require.NoError(t, err)
+
+		disChecks, err = s.GetDisabledChecks()
+		require.NoError(t, err)
+		assert.Len(t, disChecks, 1)
+	})
+
+	t.Run("disable unknown check", func(t *testing.T) {
+		sqlDB := testdb.Open(t, models.SkipFixtures, nil)
+		db := reform.NewDB(sqlDB, postgresql.Dialect, nil)
+		s, err := New(nil, nil, db)
+		require.NoError(t, err)
+		s.collectChecks(context.Background())
+
+		err = s.DisableChecks([]string{"unknown_check"})
+		require.Error(t, err)
+
+		disChecks, err := s.GetDisabledChecks()
+		require.NoError(t, err)
+		assert.Empty(t, disChecks)
+	})
+}
+
+func TestEnableChecks(t *testing.T) {
+	err := os.Setenv("PERCONA_TEST_CHECKS_FILE", "../../testdata/checks/checks.yml")
```

Comment:
`os.Setenv` is a hard no in tests – it set the environment for the whole process that limits our ability to run tests in parallel that is useful to uncover data races.

Please remove all `os.Setenv` calls there, and add `t.Parallel()`

Reply:
Removed `os.Setenv` and added `t.Parallel()` where it's possible. A lot of the tests in that file requires monopoly over `testdb`, so we can't parallel them.